connectors/src/connectors/intercom/index.ts

```diff
@@ -367,29 +409,56 @@ export async function retrieveIntercomResourcesTitles(
   connectorId: ModelId,
   internalIds: string[]
 ): Promise<Result<Record<string, string | null>, Error>> {
-  const [helpCenters, collections, articles] = await Promise.all([
+  const helpCenterIds: string[] = [];
+  const collectionIds: string[] = [];
+  const articleIds: string[] = [];
+  const teamIds: string[] = [];
+
+  internalIds.forEach((internalId) => {
+    let objectId = getHelpCenterIdFromInternalId(connectorId, internalId);
+    if (objectId) {
+      helpCenterIds.push(objectId);
+      return;
+    }
+    objectId = getHelpCenterCollectionIdFromInternalId(connectorId, internalId);
+    if (objectId) {
+      collectionIds.push(objectId);
+      return;
+    }
+    objectId = getHelpCenterArticleIdFromInternalId(connectorId, internalId);
+    if (objectId) {
+      articleIds.push(objectId);
+      return;
+    }
+    objectId = getTeamIdFromInternalId(connectorId, internalId);
+    if (objectId) {
+      teamIds.push(objectId);
+    }
+  });
```

Comment:
As `internalIds` for intercom are not just the object id but something like `intercom-collection-123-123456` the previous code was not working, and this should fix it.